### ISSUE TITLE
404 /erudio-server/web/api/tokens

### ISSUE THREAD

**Issue by marianagarciaferreira** [educacaoitajai/erudio#19]
Ola alguem poderia me ajudar com nginx?
estou tentando mas nao consigo fazer a api funcionar so da erro 404, o login aparece mas nao faz login e no console parece 404  quando tendo dar um  post em /erudio-server/web/api/tokens

**Comment by JDias** [comment]
Olá mariana,
A sra configurou?

https://github.com/educacaoitajai/erudio/blob/master/erudio-front/app/modules/main/services/erudioConfig.js.dist

Em 9 de mar de 2018 7:49 PM, "marianagarciaferreira" <
notifications@github.com> escreveu:

**Comment by marianagarciaferreira** [comment]
Eu consegui rodar a api , no entanto depois que rodei as migrations elas rodaram normal mas tento admin admin e nao vai. olhei no banco e me parece incompleto. agora estou tentando restaurar o banco e dados iniciais pelo banco que esta na pasta db, pois as migrations nao funcionam como deveriam. Estou com tudo funcionando no entanto nao consigo fazer login com admin amdin pq nao existe no banco, e quando rodei as migrations nao deu nenhum erro.

se alguem tiver o banco de dados inicial eu agradeço pq as migrations nao funcionam como deveriam

**Comment by marianagarciaferreira** [comment]
https://imgur.com/a/bj74e

depois se o pessoal precisar de ajuda para melhorar essa documentação, pq ela e nada é quase a mesma coisa.

**Comment by marianagarciaferreira** [comment]
estranho que ja rodei as migrations , ja restaurei pelo banco que e dados iniciais que vem na pasta DB, e os 2 nao fazem login. procurei o usuario admin no banco e nao o achei.

alguem saberia dizer se realmente esse banco  de dados ou migrations estao completos? 
pq rodei tanto a migration nao funcionou, restaurei o sl de estrutura e de dados iniciais e tb nao consigo fazer login porem os dados no banco foram criados.

https://imgur.com/a/wrHBw

**Comment by JDias** [comment]
Olá mariana,
Tenta usuário 76133334851 e senha 123456
Ou usuário admin e senha 123456

Quanto a documentação ela ainda não existe.

**Comment by JDias** [comment]
Qual usuario tem no banco?

Tenta ele e a senha 123456

Em 9 de mar de 2018 10:08 PM, "marianagarciaferreira" <
notifications@github.com> escreveu:

**Comment by marianagarciaferreira** [comment]
Veja ja restaurei o banco umas 10 vezes pela migration e manualmente pelo phpmyadmin

nao vejo nehum usuario no banco

**Comment by marianagarciaferreira** [comment]
https://imgur.com/a/Pl81t

olha o usuario no banco

**Comment by marianagarciaferreira** [comment]
sabe se hash da senha é md5 , se nao eu posso trocar direto no admin, mas como nao sei o hash usado
deixa tentar md5 e tentar trocar essa senha

**Comment by marianagarciaferreira** [comment]
vou continuar amanha to horas e esse login nao vai , agora so da 401 nao autorizado

**Comment by JDias** [comment]
Mariana,
Tenta admin/admin

**Comment by marianagarciaferreira** [comment]
A senha eu consegui mudar direto no banco usando MD5  Acho que meu problema é rewrite da api no nginx.
ja tentei colocar a api em subdominio , ja abri os cors no nginx.
alguem teria o rewrite para nginx?

https://imgur.com/a/XJWvn


alguem pode me ajudar a separar a api em subdominio? ou me explicar como fuinciona o rewrite no nginx?

**Comment by marianagarciaferreira** [comment]
Estou analisando e vendo mais erros
https://imgur.com/a/Io4cV

veja nas imagens que o metodo options é recebido na pai porem quando passa para o post é retornado erro 500 e veja no console o erro, apontando para linha 81 do loginController.js

vou tenrar refazer rescrever esse codigo em relação ao paths da api, da forma que esta niguem conseguira usar ou testar

**Comment by marianagarciaferreira** [comment]
mesmo erro pelo postman

https://imgur.com/a/1ELc8

**Comment by JDias** [comment]
Mariana, antes de tentar com domínios diferentes a sra tentou com a instalação "padrão"?
(A sra. está mais ou menos onde eu parei, que era tentar acessar via a API)

**Comment by marianagarciaferreira** [comment]
Sim tentei, aí a api retorna 404.

**Comment by marianagarciaferreira** [comment]
Ou vou tentar alterar o codigo fonte e ver se consigo rodar.

**Comment by JDias** [comment]
A sra conseguiu ao menos logar, sem ser pela api?
(Mas que tbm usa a api, afaik)

**Comment by marianagarciaferreira** [comment]
Não, nao funciona nada da api, estou tentando via postman mas quando o login é feito ele retorna erro 500
no entanto se eu erra o email no postman ele retorna que nao encontrou o usuario no reusltado e nao 500
https://imgur.com/a/OdRpC

**Comment by marianagarciaferreira** [comment]
vou tentar com a versao webpack agora para ver se funciona

**Comment by farribeiro** [comment]
@JDias @marianagarciaferreira Por gentileza, me passe as instruções para eu tentar reproduzir o problema, agradeço

**Comment by JDias** [comment]
Mariana, eu consegui logar seguindo as instruções básicas, então tem algo estranho ai.
Vc tem essa pagina http://erudio.sisdipe.com.br/erudio/erudio-server/web/doc ?

**Comment by JDias** [comment]
Mariana, de onde a sra verificou que sua senha md5 funcionou?

https://pt.stackoverflow.com/questions/227541/password-verify-não-retorna-o-correto

**Comment by JDias** [comment]
no meu o hash é e10adc3949ba59abbe56e057f20f883e @marianagarciaferreira 
a senha é 123456

**Comment by marianagarciaferreira** [comment]
QUeria agradecer antes de tudo a ajuda de todos !

Olha eu estou achando que tem mais haver com rewrite do nginx que qualquer outra coisa.
como vejo alguns htaccess espalhados ao longo do projeto , esotu achando que esta funcionando para apache somente.

Alguem teria testado com nginx? 

setei meu nginx para pasata root, liberei o autoindex on para conseguir listar as pastas para ver o que ocorre.
Já tenti apontar o frontend para pasta frontend, e o server para app/web/app.php.
Ja tentei deixar na raiz com as 3 pastas e colocar o nginx para ler cada uma em um dominio e nada tb. 

se alguém tiver em nginx e quiser compartilhar seu rewrite por favor. eu agradeço pois acredito que  problema seja o rewrite do nginx em relação aos paths.

Outro ponto em relação a api é os endpoints, testei pelo postmam somente a api e setei o root como web/app.php/api e tentei fazer testes rest e recebi  mesmo erro 500 nos endpoints .

o mais estranho é que trabalho com servidores a mais de 10 anos e apis rest node , php, ssr e nunca tive tanta dificuldade em colocar um projeto online como este.

Então agora so me restara remover o nginx e colocar o apache no servidor.

Mas já que foram voces que desenvolveram e eu estou achando muito confuso como isso tudo foi feito, gostaria de saber se posso contribuir refazendo essa api em Node Js e Express Js ou Laravel?
